spec/routines/create_user_spec.rb

```diff
@@ -2,37 +2,71 @@
 
 describe CreateUser do
 
-  context "when ensure_no_errors is false" do
-
-    it "fails with an error when the username is taken" do
-      FactoryGirl.create(:user, username: "bubba")
-      outcome = nil
-      expect {
-        outcome = CreateUser.call(username: "bubba", ensure_no_errors: false, state: 'activated')
-      }.to change{User.count}.by 0
-      expect(outcome.errors.has_offending_input?(:username)).to be_truthy
+  context "when all validations pass" do
+    it "creates a new user" do
+      expect{
+        CreateUser.call(username: "unclebob", title: "Mr.", first_name: "Robert", last_name: "Martin", state: "activated")
+      }.to change{User.count}.by 1
     end
-
   end
 
-  context "when ensure_no_errors is true" do
-    it "succeeds when a username is already taken" do
-      FactoryGirl.create(:user, username: "bubba")
-      outcome = nil
-      expect {
-        outcome = CreateUser.call(username: "bubba", ensure_no_errors: true, state: 'activated')
-      }.to change{User.count}.by 1
-      expect(outcome.errors).to be_empty
+  context "when validations fail" do
+    context "when ensure_no_errors is false" do
+      it "returns errors on invalid input" do
+        outcome = nil
+        expect{
+          outcome = CreateUser.call(username: "unclebob", first_name: "Bob", last_name: "Martin", ensure_no_errors: false, state: 'bogus_state!')
+        }.to_not raise_error
+
+        expect(outcome.errors).to_not be_empty
+      end
+
+      it "fails with an error when the username is taken" do
+        FactoryGirl.create(:user, username: "bubba")
+        outcome = nil
+        expect {
+          outcome = CreateUser.call(username: "bubba", ensure_no_errors: false, state: 'activated')
+        }.to change{User.count}.by 0
+        expect(outcome.errors.has_offending_input?(:username)).to be_truthy
+      end
+
     end
 
-    it "succeeds when the sanitized downcased username is already taken" do
-      FactoryGirl.create(:user, username: "Userone")
-      outcome = nil
-      expect {
-        outcome = CreateUser.call(username: "User One", ensure_no_errors: true, state: 'activated')
-      }.to change{User.count}.by 1
-      expect(outcome.errors).to be_empty
+    context "when ensure_no_errors is true" do
+      let(:invalid_input) {
```

Comment:
I re-inlined it. For some reason I had extracted but then it didn't make sense to have done that and had forgotten to re-inline it.